libs/blocks/mobile-app-banner/mobile-app-banner.js

```diff
@@ -13,8 +13,19 @@ async function getKey(product) {
   return keyMatch[0]?.key;
 }
 
+async function getECID() {
+  let ecid = null;
+  const cookiesArr = document.cookie.split(';');
+  const regex = /^AMCV_[A-F0-9]+%40AdobeOrg=MCMID\|\d+$/;
+  cookiesArr.some((el) => {
+    if (regex.test(el.trim())) [, ecid] = el.split('MCMID|');
+    return ecid;
+  });
+  return ecid;
+}
```

Comment:
You can also use `.find` to directly retrieve the matching element.
```suggestion
function getEcid() {
  const cookiesArr = document.cookie.split(';');
  const regex = /^AMCV_[A-F0-9]+%40AdobeOrg=MCMID\|\d+$/;
  const cookie = cookiesArr.find((el) => regex.test(el.trim()));
  return cookie ? cookie.split('MCMID|')[1] : null;
}
```